### PR TITLE
Add a contributing guide "CONTRIBUTING.md"

### PR DESCRIPTION
**Pull request by shacharlevy**
Fixes https://github.com/redhat-beyond/HighTeach/issues/5.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (shacharlevy) [reply CONTRIBUTING.md]: As I found "Ready for review" template is an action for automatically label swapping for draft pull requests.
It's irelevant for our current team workflow. Fixed in the second commit.

**Review by shirazyomtov — APPROVED** [review]
The file looks good to me.
Approved!

File: CONTRIBUTING.md
Comment (galitf) [inline CONTRIBUTING.md:55]: I would add that also a mentor needs to review the PR before it can be merged.

Comment (shacharlevy) [reply CONTRIBUTING.md]: Fixed :)

File: CONTRIBUTING.md
Comment (Omeramsc) [inline CONTRIBUTING.md:52]: same can be said for the commit message

Comment (shacharlevy) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md
Comment (shacharlevy) [reply CONTRIBUTING.md:53]: done

File: CONTRIBUTING.md
Comment (shacharlevy) [reply CONTRIBUTING.md:60]: done

File: CONTRIBUTING.md
Comment (shacharlevy) [reply CONTRIBUTING.md]: Ok, I removed this sentence from the PR code review section.

**Review by Yardengz — APPROVED** [review]
lgtm

**Review by natifridman — APPROVED** [review]

File: CONTRIBUTING.md
Comment (Omeramsc) [inline CONTRIBUTING.md:57]: What does "repository principle" means?

File: CONTRIBUTING.md
Comment (Omeramsc) [inline CONTRIBUTING.md:66]: maybe rephrase teammates to contributors and mentors to maintainers, as this is a public open source project.

File: CONTRIBUTING.md
Comment (Omeramsc) [inline CONTRIBUTING.md:70]: rephrase here as well

File: CONTRIBUTING.md
Comment (Omeramsc) [inline CONTRIBUTING.md:73]: oopsie :P

Comment (Omeramsc) [reply CONTRIBUTING.md:73]: but you're right, will post comments as review starting next review

**Review by Omeramsc — CHANGES_REQUESTED** [review]
Overall looking good, left some minor comment. let's get it merged today!

File: CONTRIBUTING.md
Comment (shacharlevy) [reply CONTRIBUTING.md]: It means that we should apply the principle of single responsibility. If our code is doing more than one thing we should break it into different PRs.
I reworded it and squashed it to the previous commit that  contains all the minors updates of the CONTRIBUTING.md file.

File: CONTRIBUTING.md
Comment (shacharlevy) [reply CONTRIBUTING.md]: Fixed :)

File: CONTRIBUTING.md
Comment (shacharlevy) [reply CONTRIBUTING.md]: Fixed :)

**Comment by natifridman** [comment]
Please squash the commits so we can merge it

**Pull request merged**
